### PR TITLE
Remove call to mdf5sum

### LINKED ISSUE #3944: Do not call 'md5sum' from the testsuite.
#3939 points out that we have tests that call `md5sum` via the `system()` call. This is a hidden dependency that we don't check, and consequently #3943 adds an output file for systems where there is no `md5sum` installed.

Rather, we should just avoid using `md5sum` altogether. I don't know whether `md5sum` is difficult to re-implement as a helper function in the testsuite, but suspect that to not be the case.

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #3944. The only calls to `mdf5sum` in the testsuite were in `no_flux_constraints_02` and `no_flux_constraints_03`. Admittedly, the output files are a bit large now but still not the largest in the `mpi`folder.

### CODE REVIEW

**Comment by davydden** [comment]
what if we use `std::hash` instead, applied to constraint matrix written to `ofstream`?

**Comment by davydden** [comment]
one can use `boost::hash` in order not to require `c++11` http://www.boost.org/doc/libs/1_62_0/doc/html/boost/hash.html

**Comment by masterleinad** [comment]
From [boost](http://www.boost.org/doc/libs/1_62_0/doc/html/boost/hash.html):
>[...]This hash function is not intended for general use, and isn't guaranteed to be equal during separate runs of a program - so please don't use it for any persistent storage or communication. 

and from [cppreference](http://en.cppreference.com/w/cpp/utility/hash)
>[...]The actual hash functions are implementation-dependent and are not required to fulfill any other quality criteria except those specified above.[...]
Hash functions are only required to produce the same result for the same input within a single execution of a program; this allows salted hashes that prevent collision DoS attacks. 	(since C++14)

So I guess both ideas can't really work. If we reducing the number of refinements, I am not sure whether the behavior the tests are written for is still tested.

**Comment by davydden** [comment]
That's a pity

**Comment by bangerth** [comment]
I'm almost tempted to just take the reference implementation of md5 and stick it into namespace `Utilities` :-) It is available here: https://tools.ietf.org/html/rfc1321 The alternative would of course be to just implement *some* kind of checksum -- it doesn't have to be particularly good.

But for the moment, I think this patch is ok. If nobody raises any further objections within a day, feel free to merge, @masterleinad .

**Comment by tjhei** [comment]
Yeah, this was an ugly hack I did to check that the contraints are handled correctly. I don't like that we now add 10k lines of reference output but it is at least better than using md5sum directly, so :+1:

**Comment by tjhei** [comment]
update: I will put a simple checksum calculation routine into ``tests.h`` and call that one instead (tomorrow or so).

**Comment by masterleinad** [comment]
One other try. I just picked some implementation that seems to be used at multiple places online. I guess the copyright should be fine. Since we are just using this in `tests` I let it live only there so far.
I am happy to drop the first commit, if we can agree on this.

File: tests/md5/md5.cc [hunk 1]
```diff
@@ -0,0 +1,374 @@
+/* MD5
+ converted to C++ class by Frank Thilo (thilo@unix-ag.org)
+ for bzflag (http://www.bzflag.org)
```
Comment (tjhei) [inline tests/md5/md5.cc]: the licensing of this is really unclear to me. Do you mind if I show you my suggestion?

Comment (davydden) [inline tests/md5/md5.cc]: if we have a self-written implementation, i would put it into `Utilities::`. I would actually also move `rand` there. `rand` can be used to add function to deal.II vector to initialize with random numbers, maybe hashes can also be used somewhere later.

**Pull request closed without merging**